docs-src/typescript/chapter-durable-execution/backgroundcheck-replay-inspecting-the-non-deterministic-error.md

```diff
@@ -0,0 +1,78 @@
+---
+id: backgroundcheck-replay-inspecting-the-non-deterministic-error
+title: Intrinsic non-deterministic logic
+sidebar_label: intrinsic-non-deterministic-logic
+description: This kind of logic prevents the Workflow code from executing to completion because the Workflow can take a different code path than the one expected from the Event History.
+tags:
+- tests
+- replay
+- event history
+---
+
+If you run the `BackgroundCheckNonDeterministic` Workflow enough times, eventually you will see a Workflow Task failure.
```

Comment:
I'm curious... How do you plan to implement `BackgroundCheckNonDeterministic` in TS?

Reply:
Really good question. Faced something similar in Python.
Would it be best to ignore this section due to the V8 isolation?